File: linkedList.java
public class linkedList {
    static class Node {
        //node definition
        int data; //stores value
        Node next; //store refeference to the next node

        Node(int value) {
            data = value;
            next = null;
        }  
    }
    static Node head = null;

    // Insert at front
    static void insertFront(int value) {

        Node newNode = new Node(value);
        newNode.next = head;
        head = newNode;
    }

    // Delete from front
    static void deleteFront() {

        if (head == null) {
            System.out.println("List is empty");
            return;
        }

        head = head.next;
    }

    // Display the linked list
    static void display() {

        Node temp = head;

        while (temp != null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("NULL");
    }

    public static void main(String[] args) {
        insertFront(10);
        insertFront(20);
        insertFront(30);
        display();

        deleteFront();
        display();
    }


}

File: linkedList.pas
program LinkedList;

type
  NodePtr = ^Node; //pointer
  Node = record
    data: Integer;
    next: NodePtr;
  end;

var
  head, temp: NodePtr;

// insert at front
procedure InsertFront(var head: NodePtr; value: Integer);
var
  newNode: NodePtr;
begin
  new(newNode);
  newNode^.data := value;
  newNode^.next := head;
  head := newNode;
end;

// delete from front
procedure DeleteFront(var head: NodePtr);
var
  temp: NodePtr;
begin
  if head = nil then
    writeln('List is empty')
  else
  begin
    temp := head;
    head := head^.next;
    dispose(temp);  //free memory
  end;
end;

procedure Display(head: NodePtr);
begin
  if head = nil then
    writeln('List is empty')
  else
  begin
    while head <> nil do
    begin
      write(head^.data, ' -> ');
      head := head^.next;
    end;
    writeln('NULL');
  end;
end;

begin
  head := nil;

  InsertFront(head, 10);
  InsertFront(head, 20);
  InsertFront(head, 30);
  Display(head);

  DeleteFront(head);
  Display(head);
end.
